### PR TITLE
Adds a new way to get research points.

### PR DESCRIPTION
**Pull request by lolman360**
## About The Pull Request

tldr: vr away missions but they give points

Adds a new way to get research points. Scientists log into a special VR sleeper, which takes them to cyberspace. In cyberspace they have a "base" with some weapons, equipment, a vendor and a protolathe linked to the science techweb. To get points they must raid other faction's "bases" and destroy their "node" to steal their research (i.e get points)  These bases are defended by turrets, faction simplemobs, etc. These drop things like materials (to use in the protolathe), point cards (to buy things from the vendor) as well as the occasional new weapon or something. More challenging bases give more points and rewards, while easier ones give less. Dying as a VR instance in cyberspace will delete all the items you had on you at the time.

## Checklist

- [ ] point nodes
- [ ] maps
- [ ] simplemobs
- [ ] virtual techweb linking

## Why It's Good For The Game

feature man good, rushing bomb in 6 minutes and getting 50k points bad

## Changelog
:cl:
add: Added new way to get points.
/:cl:

### CODE REVIEW

**Comment by Trilbyspaceclone** [comment]
This isnt Tetris

**Comment by Linzolle** [comment]
Trilby are you suggesting we play tetris to generate research because I would generate so many points

File: code/modules/research/machinery/pointgiver.dm [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/obj/machinery/rnd/pointgiver
```
Comment (Ghommie) [inline code/modules/research/machinery/pointgiver.dm:1]: This could as well just be a /obj/structure subtype in the current state.
But I aknowledge it's pretty WIP at the moment and perhaps you may have some design in mind that justifies them being a machinery/rnd subtype.
Also they'll need an icon_state.

File: code/modules/research/machinery/pointgiver.dm [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  name = "virtual technology node"
+  desc = "A virtual representation of stored research, ideas and knowledge. This one belongs to [factionname] and will give [reward] points if destoyed."
```
Comment (Ghommie) [inline code/modules/research/machinery/pointgiver.dm]: I doubt squared parenthesis values will do the trick here.

You should either set it up on `Initialize()`, or add this to its `examine()` :
```
  to_chat(user, <span class='notice'>This one belongs to [factionname] and will give [reward] points if destoyed.</span>)
```

**Comment by Ludox235** [comment]
if you gonna copy lifeweb atleast make it good

**Comment by AldenBluetenberger** [comment]
bro if you copy lifeweb randy will castrate you lmao

**Comment by Arturlang** [comment]
Shhhhhhh, its a PARODY.

**Comment by lolman360** [comment]
not stale, busy irl
